### ISSUE TITLE
need option for specifying date format 

### ISSUE THREAD

**Issue by barrybecker4** [databricks/spark-csv#279]
I noticed these comments in the code:
// TODO(hossein): would be good to support other common timestamp formats
case _: TimestampType => Timestamp.valueOf(datum)
// TODO(hossein): would be good to support other common date formats
case _: DateType => Date.valueOf(datum)

It would be really nice to be able to specify the format of date columns when reading in a CSV file.
For example, something like this:

df = sqlContext.read
                .format("com.databricks.spark.csv")
                .option("inferSchema", "false") 
                .option("dateFormat", "mm/dd/yyyy hh:mm:ss")
                .schema(dfSchema)
                .load(dataFile)

Currently the only format allowed is (yyyy-mm-dd hh:m:ss")
Is there a workaround for this besides just changing the data before hand? I suppose you could read all date columns as string then operate on the dataframe to change them all to dates, but it would be pretty tedious.

**Comment by HyukjinKwon** [comment]
I think this is a duplicate of https://github.com/databricks/spark-csv/issues/262?

**Comment by barrybecker4** [comment]
Yes, it's basically the same as #262. Sorry I did not see it. One can be duped to the other.

**Comment by HyukjinKwon** [comment]
@barrybecker4 Could we maybe close this? This is just fixed in #280.

**Comment by barrybecker4** [comment]
closing.

**Comment by barrybecker4** [comment]
I realize that this issue has been closed, but I had two questions.
1) Assuming that the reading and writing of a file is happening in parallel, I think it would be safer to use Joda's DateTimeFormat (which is threadsafe) instead SimpleDateFormat (which is not threadsafe). Is this a concern?
2) Is the dateFormat used when writing? It appears that it is not. I will open a separate issue for this.
Here is what I am doing and it does not write in the specified format:
repartitionDf.write.format("com.databricks.spark.csv")
        .option("delimiter", "\t")
        .option("header", "false")
        .option("nullValue", "?")    // this worked
        .option("dateFormat", "MM/dd/yyyy HH:mm:ss")
        .save(tempFileName)
The file has dates like "2012-01-03 09:12:00.0" but it should be 03/01/2012 09:12:00"
